ModularContent/Fields/Post_List.php

```diff
@@ -559,84 +570,127 @@ protected static function get_p2p_filtered_ids( $connection_id, $post_ids ) {
+		$blueprint['hidden_fields']    = $this->hidden_fields;
+		$blueprint['post_type']        = [];
 		foreach ( $this->post_type_options() as $pto ) {
-			if ( !is_object( $pto ) ) {
+			if ( ! is_object( $pto ) ) {
 				$pto = get_post_type_object( $pto );
 			}
-			$blueprint[ 'post_type' ][] = [
+			$blueprint['post_type'][] = [
 				'value' => $pto->name,
 				'label' => $pto->label,
 			];
 		}
-		$blueprint[ 'filters' ] = [ ];
-		$taxonomy_options = [ ];
+		$blueprint['filters'] = [];
+		$taxonomy_options     = [];
 		foreach ( $this->taxonomy_options() as $tax ) {
-			$taxonomy = get_taxonomy( $tax );
+			$taxonomy           = get_taxonomy( $tax );
 			$taxonomy_options[] = [
 				'value'       => $tax,
 				'label'       => $taxonomy->label,
 				'filter_type' => 'taxonomy',
 				'post_type'   => Util::get_post_types_for_taxonomy( $tax ),
 			];
 		}
-		if ( !empty( $taxonomy_options ) ) {
-			$blueprint[ 'filters' ][] = [
+		if ( ! empty( $taxonomy_options ) ) {
+			$blueprint['filters'][] = [
 				'label'   => $this->get_string( 'label.taxonomy' ),
 				'options' => $taxonomy_options,
 			];
 		}
 
-		$p2p_options = [ ];
+		$p2p_options = [];
 		foreach ( $this->p2p_options() as $relationship_id => $relationship ) {
 			$post_types_for_p2p = \ModularContent\Util::get_post_types_for_p2p_relationship( $relationship );
-			$p2p_options[] = [
+			$p2p_options[]      = [
 				'value'       => $relationship_id,
 				'label'       => Util::get_p2p_relationship_label( $relationship ),
 				'filter_type' => 'p2p',
 				'post_type'   => $post_types_for_p2p,
 			];
 		}
-		if ( !empty( $p2p_options ) ) {
-			$blueprint[ 'filters' ][] = [
+		if ( ! empty( $p2p_options ) ) {
+			$blueprint['filters'][] = [
 				'label'   => $this->get_string( 'label.relationship' ),
 				'options' => $p2p_options,
 			];
 		}
 
-		$blueprint[ 'filters' ][] = [
+		$blueprint['filters'][] = [
 			'value'       => 'date',
 			'label'       => $this->get_string( 'label.date' ),
 			'filter_type' => 'date',
 			'post_type'   => Util::get_post_types_for_date(),
 		];
 
-		$blueprint[ 'taxonomies' ] = [ ];
+		$blueprint['taxonomies'] = $this->get_taxonomy_terms();
 
-		foreach ( $this->taxonomy_options() as $taxonomy_name ) {
-			$terms = get_terms( $taxonomy_name, [ 'hide_empty' => false ] );
-			$options = [ ];
-			foreach ( $terms as $term ) {
-				$term_name = self::build_hierarchical_term_name( $term );
-				$options[] = [
-					'value' => $term->term_id,
-					'label' => html_entity_decode( $term_name, ENT_QUOTES | ENT_HTML401 ), // it will be re-encoded later
-				];
+		return $blueprint;
+
+	}
+
+	/**
+	 * Get and cache all taxonomy terms.
+	 *
+	 * @return array
+	 */
+	private function get_taxonomy_terms() {
+		global $_wp_using_ext_object_cache;
+
+		$transient = 'panels_tax_terms';
+
+		// Store current object cache
+		$object_cache = $_wp_using_ext_object_cache;
+
+		// Force transients to use the database
+		$_wp_using_ext_object_cache = null;
```

Comment:
This is a really great solution. I wonder if there should be a filter to determine if we allow object cache usage or always force DB, but I link I lean on the side of leaving it as-is.